### ISSUE TITLE
EFCorePowerTools Not seeing dependencies

### ISSUE THREAD

**Issue by tburger** [ErikEJ/EFCorePowerTools#143]
Describe what is not working as expected.
I have a SQL table with several FK dependencies to other tables.  Many of them show up fine but a few do not show.  They do not show in the SSMS show dependencies tool AND are not being picked up by EFCorePowerTools BUT they do show when I run the following: EXEC sp_MSdependencies N'Item.Item', null, 1053183 .  I can not figure out anything that is different about these FK relationships and the ones that work.  

### Steps to reproduce
Include a complete description that I can redo to reproduce the issue.
Maybe if I understood what sql command that EFCorePowerTools is using to reverse engineer the Dependencies, I could figure out how these FK relationships are different than others.  I have looked at size of table name and size of FK relationship name... it is nothing obvious 

### Further technical details
EF Core Power Tools version: (found in About dialog - blue questionmark icon on context menu)
Version 1.0667
Database engine: (SQlite, SQL Compact, SQL Server, Postgres)
SQL Server 2017 Enterprise
Visual Studio version: (e.g. Visual Studio 2017 15.7)
Visual Studio 2017

**Comment by ErikEJ** [comment]
Hi, the SQL command to get the foreign keys is available here: https://github.com/aspnet/EntityFrameworkCore/blob/master/src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs#L900

**Comment by tburger** [comment]
Thank you.  I thought I was on a role with that but then I realized that is only the depends on not the things that depend on it.  My problem is that EFCorePowerTools is not making a relationship from item to 3 different tables that have a foreign key but it does for that are defined the same way.  The same problem that is causing EFCorePowerTools to not map it also seems to show up in SSMS.

The SSMS also does not show the relationship back so that is the easiest way for me to describe the issue:
Here you will see that PurchaseOrderDetailHistory depends on Item
[cid:image002.jpg@01D48F40.53CEAEC0]

So Item should show that PurchaseOrderDetailHistory has a dependency on it
[cid:image004.png@01D48F40.0CCA9930]

**Comment by ErikEJ** [comment]
I cannot see any images, you need to upload on Github web site. But it would be much better with a script to create your schema...

**Comment by tburger** [comment]
Attached are both images and a script to recreate the db schema.  You will need to create a new db then run the script to create the tables and relationships.  

[testdb.txt](https://github.com/ErikEJ/EFCorePowerTools/files/2660316/testdb.txt)
![sql1](https://user-images.githubusercontent.com/629105/49694484-9163ea80-fb50-11e8-8f15-a281075c002c.jpg)
![sql2 jpg](https://user-images.githubusercontent.com/629105/49694485-93c64480-fb50-11e8-8f2d-e28203cf6cd9.png)

**Comment by tburger** [comment]
Here is a pic of what comes up with EXEC sp_MSdependencies N'Item.Item', null, 1053183 
All of the highlighted tables are not showing up in the Gui as dependent on Item.  I assume this also means they are not mapping in EFCorePowerTools Reverse engineering.  I only know the PurchaseOrderDetailHistory is not.  The db is still changing frequently so we are regenerating often.

![tawx5](https://user-images.githubusercontent.com/629105/49694609-12bc7c80-fb53-11e8-8a90-da89837c42c0.png)

**Comment by ErikEJ** [comment]
Not sure how sp_depends and SSMS interperts "depends", but the EF Core Power Tools requires a FK to be defined - what I am saying is that looking for dependencies that are missing in the UI is an issue that does not belong here.

 The results from the sql query I linked to are much more interesting!

**Comment by ErikEJ** [comment]
Thanks for the sample database. I do not understand what your exact issue actually is!

>>any of them show up fine but a few do not show. 

Show up where?

>> are not being picked up by EFCorePowerTools 

What does "not being picked up" mean?

**Comment by tburger** [comment]
My apologies..  This all started over PurchaseOrderDetailHistories not being added to the Item model when EFCorePowerTools was used to generate the model.  I used the SQL gui to show her what she was missing in her check and she showed me that it was not being generated and thus why she couldn't do what I showed.  Some how in all of this, either I never tried it on my box or I have shook a tree loose and it is working for me.  I will check with her tomorrow and see if it works for her also now or we can isolate that problem.   I am sorry for waisting your time.
